src/main/java/uk/gov/pay/ledger/agreement/entity/AgreementsFactory.java

```diff
@@ -32,4 +45,10 @@ public PaymentInstrumentEntity createPaymentInstrument(EventDigest eventDigest)
         entity.setEventCount(eventDigest.getEventCount());
         return entity;
     }
+
+    private AgreementStatus getStatus(EventDigest eventDigest) {
+        var eventType = eventDigest.getMostRecentSalientEventType().get();
+
+        return EVENT_TYPE_AGREEMENT_STATE_MAP.get(eventType);
```

Comment:
We should never just call `get` on an Optional without checking whether the value is present, we should either return a default value or throw an exception. In this case, I think an exception would make sense as we would always expect one to be returned

```suggestion
        return eventDigest.getMostRecentSalientEventType().map(eventType -> EVENT_TYPE_AGREEMENT_STATE_MAP.get(eventType)).orElseThrow(() -> new RuntimeException("No salient event found for event digest for agreement with external ID: [xxxx]"));
```